Mettre à jour le barème des AF

MattiSG opened the pull request:
En tant que bénéficiaire potentiel faisant une simulation au 1er juillet 2015,
Je peux bénéficier du nouveau barème,
Pour avoir une simulation à jour

- [ ] Twitter 
- [ ] Wiki

Commit:
Possibilité de neutral total revenus ASS

Commit:
Neutral totale rev Stage + ARE

cbenz committed:
Merge pull request #276 from sgmap/ass-neutral-are

ASS - Neutralisation totale ARE et revenus stage si interrompus

Commit:
Modulation des AF juillet 2015

Commit:
Modulation à 1 avant juillet 2015

MattiSG committed:
Add tests for AF 2015-07

MattiSG committed:
Fix complement_degressif

MattiSG committed:
Refactor complement_degressif

MattiSG committed:
Refactor complement_degressif

MattiSG committed:
Correct taux modulation AF

Source : https://www.caf.fr/aides-et-services/s-informer-sur-les-aides/petite-enfance/les-allocations-familiales-af-0

MattiSG commented:
Il faut modifier la base de ressources sur laquelle la modulation des AF se base : on utilise actuellement `br_pf`, qui correspond au revenu imposable **après abattements** (notamment des frais réels, 10%), alors que la nouvelle base ressource des AF est constituée des revenus imposables (« nets sur lesquels l'impôt est calculé » — [R532-3 CSS](http://legifrance.gouv.fr/affichCodeArticle.do;jsessionid=1B4887DFDFBD506B25B643530FD9BA38.tpdila11v_1?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000006750894&dateTexte=&categorieLien=cid)) **sans abattements**.

MattiSG commented:
A priori, la base de ressources à utiliser est [`rev_cat_tspr`](http://legislation.openfisca.fr/variables/rev_cat_tspr).

MattiSG commented:
Non, `rev_cat_tspr` applique aussi un abattement (73000€ de "revenus d'activité connus" => 65700€). Pourtant, la définition « Revenu catégoriel - Traitements, salaires, pensions et rentes » semble correspondre au [R532-3 CSS](http://legifrance.gouv.fr/affichCodeArticle.do;jsessionid=1B4887DFDFBD506B25B643530FD9BA38.tpdila11v_1?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000006750894&dateTexte=&categorieLien=cid).

@benjello une idée de quelle formule utiliser ?

MattiSG commented:
@Anne-LaureM ajoute la définition de « revenus nets catégoriels » d'après [les impôts](http://www.impots.gouv.fr/portal/dgi/public/popup;jsessionid=G4URLGBRJXQUFQFIEIPSFEY?docOid=documentstandard_2417&espId=0&typePage=cpr02&hlquery=revenu%20net&temNvlPopUp=true) :

> Pour les traitements, les salaires, les pensions et les retraites, le revenu net catégoriel est obtenu en appliquant sur votre revenu imposable une déduction de 10 % puis un abattement de 20 %.

73 * 0,9 = 65,7 => ok, 10% abattus
65,7 * 0,8 = 52,56 => donc `rev_cat_tspr` ne prend pas en compte le second abattement donné par les impôts… et je ne sais toujours pas laquelle de ces trois valeurs prendre en compte pour les AF  >.<

MattiSG commented:
@benjello fait remarquer que l'abattement de 20% n'existait que jusqu'en 2006… et que la référence citée plus haut date de 2006  ;)  Il ne subsiste donc aujourd'hui a priori que l'abattement de 10% (moins les pensions alimentaires et des abattements fiscaux [apparemment](http://vosdroits.service-public.fr/particuliers/R10278.xhtml))… qui devraient être pris en compte dans le `br_pf`.

(au passage, on n'est pas les seuls à s'être [arrachés les cheveux](http://kevin.polisano.free.fr/?p=569))

@adrienpacifico, pourrais-tu stp confirmer :

1. Notre interprétation de la base ressources prise en compte pour les AF (revenus nets catégoriels => imposable abattu de 10% moins pensions alimentaires…).
2. Qu'OpenFisca calcule cette base dans [`br_pf`](http://legislation.openfisca.fr/variables/br_pf).

(et tant qu'à faire, si tu veux revoir le code de cette PR, tu es plus que bienvenu !  ;)  )

adrienpacifico commented:
J'ai jeté un coup d'oeil rapide sur la PR, ce n'est pas exactement ce que j'ai codé car je n'avais pas la même problématique (pas de dom-tom etc).

Pour la br_pf le simulateur de modulation de la CNAF est existe, j'avais essayé de comprendre le calcul exact en décembre et je m'étais un peu arraché les cheveux (Loi + Code :) ).

http://www.caf.fr/actualites/2015/allocations-familiales-le-simulateur

En prenant le simulateur  il est indiqué ressource imposable du foyer N-2 (2013) est la variable de référence.

Pour retomber sur mes pates avec le seuil de 67 140 (deux enfants) : 
CSG-déductible+CRDS =1- (2.4+0.5) = (100% - 2.9%) = 0.971
abatement pour frais professionnel = 90%

0.9/0.971 = 0.92687951

6000 euros mensuel net = 72 000 annuel
si je prends 72000 euros de salaire_net *0.92687951 = 66735
Le plafond sur le simulateur de la caf pour deux enfants démarre à 66740, ce qui matche à priori très bien (pareil pour les enfants à 500 euros, le deuxième seuil à 8000 net mensuel,  etc).

Je suppose donc que la caf ne fait qu'un abattement de 0.9 sur les revenus imposables.

Par contre je vais garder la conversation en tête pour les références législatives (c'est un peu painfull de chercher de 0) -- et surveiller le repo sgmap car je ne savais pas qu'il se passait des choses ici :).

Une note de L'IDEP devrait sortir très bientôt (elle est finie, elle attends d'être publiée) sur la réforme de modulation des allocations familiales.

Commit:
Remove unused variable

Commit:
Clean default values

Commit:
Update salaire_de_base metadata

Commit:
Add reference to inversion_revenus.salaire_de_base

Commit:
Remove unused variable

MattiSG commented:
:heart: Merci @adrienpacifico !!

On est donc a priori bons, puisque les "revenus d'activité" que l'on capture dans Mes Aides sont déjà un montant imposable auquel il suffit donc d'appliquer l'abattement de 10%, sans jonglage avec la CSG et la CRDS.

Je mets à jour les tests (`/= 0,9`) et je reviens.

MattiSG committed:
Fix tests for AF-2015

Feed imposable rather than net
Remove values that were not expected to be tested

MattiSG commented:
On a encore des soucis (généralement le montant est doublé par rapport à ce qu'il devrait être). Je redonne le lead à @fpagnoux, avec des tests.

cbenz committed:
Merge pull request #277 from adrienpacifico/next

change calculus of ass to caluculate_add in mini

MattiSG commented:
[Implémentation alternative](https://github.com/adrienpacifico/Modulation_Allocations_Familiales/blob/master/modulation_allocations_familiales/reforms/af_modulation.py) par @adrienpacifico.

fpagnoux commented:
Il faut vérifier qu'il y a bien dépassement de plafond avant d'atttribuer le complément dégressif. Je pense que le facteur 2 vient de là.

Commit:
Complément degressif seulement si dépassement plafond

fpagnoux commented:
Par ailleurs beaucoup de tests ludwig ont pour période de calcul le mois de juin, et donc ne prennent pas en compte la modulation.

MattiSG commented:
Utilise les tests OpenFisca que j'ai joints à cette PR, ils sont à jour  ;)

fpagnoux commented on openfisca_france/model/prestations/prestations_familiales/af.py:
```diff
@@ -231,7 +237,43 @@ def function(self, simulation, period):
+            (depassement_plafond2 > 0) * depassement_plafond2
+        ) / 12
+
+        af = af_base + af_majo + af_forf  # TODO: af_forf est-il inclus ?  cf. dernier paragraphe de http://legifrance.gouv.fr/affichCodeArticle.do;jsessionid=1B4887DFDFBD506B25B643530FD9BA38.tpdila11v_1?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000030680318&dateTexte=20150624&categorieLien=id#LEGIARTI000030680318
```
En lisant les paragraphes II des deux articles suivants : 
[Article D521-1](http://legifrance.gouv.fr/affichCodeArticle.do;jsessionid=1B4887DFDFBD506B25B643530FD9BA38.tpdila11v_1?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000030680318&dateTexte=20150624&categorieLien=id#LEGIARTI000030680318)
[Article D521-2](http://legifrance.gouv.fr/affichCodeArticle.do;jsessionid=1B4887DFDFBD506B25B643530FD9BA38.tpdila11v_1?cidTexte=LEGITEXT000006073189&idArticle=LEGIARTI000030680324&dateTexte=20150624&categorieLien=id#LEGIARTI000030680324)
j'ai l'impression qu'il y a deux compléments dégressifs indépendants, un pour l'allocation familiale (+ majoration), l'autre pour l'allocation forfaitaire. 

Mais traiter les deux compléments indépendamment désavantage le demandeur : pour chaque euro ajouté à sa base ressource, il perd un euro sur chaque complément, donc 2€ au total. Et donc, globalement, il n'a pas intérêt à augmenter ses ressources, ce qui n'est pas dans l'esprit du complément. 

Je pense que c'est un point à éclaircir auprès de quelqu'un qui connaît les détails.

fpagnoux replied:
@Anne-LaureM vient de me signaler un  document qui explique comment gérer le cas. Je regarde plus en détails.
Merci ;)

MattiSG replied:
J'interprète comme toi, mais reste à savoir si le forfaitaire est cumulable avec les autres, peut-être que ça ne peut jamais arriver.

fpagnoux replied:
Demi-réponse : on ne peut mathématiquement pas être éligible au deux compléments dégressifs. Mais la mauvaise nouvelle, c'est que les enfants pris en compte pour les calculs respectifs des AF "normales" et forfaitaires ne se basent pas sur les mêmes plafonds (car les enfants ne sont pas pris en compte de la même manière).
Cela va complexifier le code un peu plus (pas de `af_taux_modulation` unique)

fpagnoux replied:
Après débat, il semble que la situation de l'éligibilité aux deux compléments soit possible, mais uniquement dans des cas assez spécifiques, de type famille très nombreuse (e.g. 8 enfants de moins de 20 ans, plus 1 de 20 an), avec des ressources très légèrement supérieures à un des plafonds. L'impact n'est pas gigantesque (40€ par mois max).
On a de toute façon implémenté comme le prévoit les textes et le doc CAF.

Commit:
Introduce af_forf_taux_modulation
Rename complement_degressif to af_complement_degressif
Introduce af_forf_complement_degressif

Commit:
Relative margin in tests

fpagnoux commented:
Tous les tests ludwig en france métropolitaine sont verts.

Certains tests sont rouges pour les DOM, mais c'est probablement issu d'une erreur sur le site de la CAF, qui indique des plafonds différents pour les DOM et la métropole, alors que les textes de loi et la notice envoyée aux CAF ne font pas état de cette différence.

Concernant la possible cumulation des deux compléments dégressifs, je ne suis pas d'accord avec la CAF qui indique que cette situation est impossible.  Mais cela n'a pas d'impact sur l'implémentation : on suit leurs instructions, même si ça désavantage le demandeur dans un cas bien spécifique.

Commit:
Introduce af_forf_nbenf
Refactor af_forf

Commit:
Clean print

Commit:
Simplif calcul du plafond

Commit:
Maj tests yaml

fpagnoux commented:
Reste  à faire
- [ ] Mettre à jour les tests sur le serveur de prod en faisant tourner le script de migration
- [ ] Déploiement
